### ISSUE TITLE
Improvment: Reduce boilerplate code in startup

### ISSUE THREAD

**Issue by msallin** [BEagle1984/silverback#85]
Currently we need to get IHostApplicationLifetime & BusConfigurator in the Configure of the startup, to do the following.

```
IBroker broker = busConfigurator.Connect();
applicationLifetime.ApplicationStopping.Register(broker.Disconnect);
```

Can this be done behind the scene, that we can get rid of the two overloads and the two lines of code.

**Comment by BEagle1984** [comment]
This is in my backlog already, most probably for release 3.0. 😉

**Comment by BEagle1984** [comment]
This optimization is included in v3.0.0 (currently in beta, latest being [v3.0.0-beta.streaming.5](https://github.com/BEagle1984/silverback/releases/tag/v3.0.0-beta.streaming.5).

An hosted service is now being used to automatically connect and disconnect the message broker and the configuration has been tweaked to be more concise and (hopefully) straightforward.

https://github.com/BEagle1984/silverback/blob/05a0725ec430ef99413860928b335dae73a7da9f/samples/Kafka/BinaryFileStreaming.Consumer/Startup.cs#L16-L29

https://github.com/BEagle1984/silverback/blob/05a0725ec430ef99413860928b335dae73a7da9f/samples/Kafka/BinaryFileStreaming.Consumer/EndpointsConfigurator.cs#L12-L45

Note 1: `IEndpointConfigurator`  is now the preferred and default way to configure the endpoints but they can still be configured in the `Startup` directly, but now it is done in the `ConfigureService`, using `AddEndpoints` instead of `AddEndpointsConfigurator`. Proper documentation will follow...
Note 2: The actual endpoint configuration is still subject to change in the near future. (Probably adding some builders/methods to help with settings the properties like `ErrorPolicy` and similar.)

**Comment by msallin** [comment]
Cool! Ty!
